sympy/stats/tests/test_continuous_rv.py

```diff
@@ -147,12 +151,144 @@ def test_characteristic_function():
     Y = Normal('y', 1, 1)
     cf = characteristic_function(Y)
     assert cf(0) == 1
-    assert simplify(cf(1)) == exp(I - S(1)/2)
+    assert cf(1) == exp(I - S(1)/2)
 
     Z = Exponential('z', 5)
     cf = characteristic_function(Z)
     assert cf(0) == 1
-    assert simplify(cf(1)) == S(25)/26 + 5*I/26
+    assert cf(1).expand() == S(25)/26 + 5*I/26
+
+def test_moment_generating_function():
+    t = symbols('t', positive=True)
+
+    # Symbolic tests
+    a, b, c = symbols('a b c')
+
+    mgf = moment_generating_function(Beta('x', a, b))(t)
+    assert mgf == hyper((a,), (a + b,), t)
+
+    mgf = moment_generating_function(Chi('x', a))(t)
+    assert mgf == sqrt(2)*t*gamma(a/2 + S(1)/2)*\
+        hyper((a/2 + S(1)/2,), (S(3)/2,), t**2/2)/gamma(a/2) +\
+        hyper((a/2,), (S(1)/2,), t**2/2)
+
+    mgf = moment_generating_function(ChiSquared('x', a))(t)
+    assert mgf == (1 - 2*t)**(-a/2)
+
+    mgf = moment_generating_function(Erlang('x', a, b))(t)
+    assert mgf == (1 - t/b)**(-a)
+
+    mgf = moment_generating_function(Exponential('x', a))(t)
+    assert mgf == a/(a - t)
+
+    mgf = moment_generating_function(Gamma('x', a, b))(t)
+    assert mgf == (-b*t + 1)**(-a)
+
+    mgf = moment_generating_function(Gumbel('x', a, b))(t)
+    assert mgf == exp(I*b*t)*gamma(-a*t + 1)
```

Comment:
Shouldn't the exponential term have been `exp(b*t)`?

Reply:
AFAIK, the PDF of Gumbel Distribution hasn't been corrected in master. It was merged in the release branch. Am I right @sidhantnagpal?

Reply:
> PDF of Gumbel Distribution hasn't been corrected in master

It has the release branches are merged back into master after the release. You can see the version of the method in the branch (and even check the git history to see that the pdf of Gumbel was changed).
https://github.com/sympy/sympy/blob/6f7d053881472f426cf13e6793d782f43b4afa9f/sympy/stats/crv_types.py#L1575-L1578

The issue seems to be in the definition of `Gumbel._moment_generating_function`:
https://github.com/sympy/sympy/blob/6f7d053881472f426cf13e6793d782f43b4afa9f/sympy/stats/crv_types.py#L1587-L1588

Reply:
> PDF of Gumbel Distribution hasn't been corrected in master

It has, the release branches are merged back into master after the release. You can see the version of the method in the branch (and even check the git history to see that the pdf of Gumbel was changed).
https://github.com/sympy/sympy/blob/6f7d053881472f426cf13e6793d782f43b4afa9f/sympy/stats/crv_types.py#L1575-L1578

The issue seems to be in the definition of `Gumbel._moment_generating_function`:
https://github.com/sympy/sympy/blob/6f7d053881472f426cf13e6793d782f43b4afa9f/sympy/stats/crv_types.py#L1587-L1588